src/pages/library/DataBrowserDetails.js

```diff
@@ -194,17 +195,21 @@ export const SidebarComponent = ({ dataObj, id }) => {
         ])
       ]),
       h(ButtonPrimary, {
-        disabled: true,
-        tooltip: 'We are currently working on the link to workspace feature which will be available soon.',
+        disabled: dataObj.access !== datasetAccessTypes.GRANTED,
+        tooltip: dataObj.access === datasetAccessTypes.GRANTED ? '' : uiMessaging.controlledFeature_tooltip,
         style: { fontSize: 16, textTransform: 'none', height: 'unset', width: sidebarButtonWidth, marginTop: 20 },
         onClick: () => {
           Ajax().Metrics.captureEvent(`${Events.catalogWorkspaceLink}:detailsView`, {
             id,
             title: dataObj['dct:title']
           })
-          importDataToWorkspace([dataObj])
+          importDataToWorkspace(dataObj, () => setPreparingExport(true))
         }
-      }, ['Link to a workspace']),
+      }, [preparingExport ?
```

Comment:
I wonder if this may stop working when we upgrade to React 18 (currently using `17.0.2`) due to [automatic batching](https://github.com/reactwg/react-18/discussions/21). Probably not based on the examples on that page but I figured I'd bring it up.